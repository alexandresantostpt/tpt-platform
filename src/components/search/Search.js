import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { i18n } from '@i18n'

import Icon from '@components/icon/Icon'

const StyledSearch = styled.div`
    margin-right: 1rem;
    position: relative;
    width: 77%;

    .icon-search {
        color: #858899;
        position: absolute;
        right: 0.3rem;
        top: 0.7rem;
    }
`

const StyledSearchInput = styled.input`
    border: 1px solid #f2f2f2;
    border-radius: 10px;
    box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.16);
    box-sizing: border-box;
    color: ${({ theme }) => theme.text};
    font-size: 1rem;
    height: 2.15rem;
    opacity: 0.45;
    padding: 0.6875rem 1rem;
    width: 100%;

    &:-moz-placeholder {
        color: rgba(0, 0, 0, 0.45);
        text-align: center;
    }

    &:-ms-input-placeholder {
        color: rgba(0, 0, 0, 0.45);
        text-align: center;
    }

    &::-moz-placeholder {
        color: rgba(0, 0, 0, 0.45);
        text-align: center;
    }

    &::-webkit-input-placeholder {
        color: rgba(0, 0, 0, 0.45);
        text-align: center;
    }

    &:focus {
        outline: none;
    }
`

const Search = ({ onType, placeholder }) => (
    <StyledSearch>
        <StyledSearchInput
            onChange={({ target }) => onType(target.value)}
            placeholder={placeholder || i18n.t('placeholders.search')}
        />
        <Icon fontSize="0.9rem" icon={'search'} />
    </StyledSearch>
)

Search.defaultProps = {
    placeholder: ''
}

Search.propTypes = {
    onType: PropTypes.func.isRequired,
    placeholder: PropTypes.string
}

export default Search
