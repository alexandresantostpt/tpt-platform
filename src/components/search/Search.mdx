---
name: Search
route: /search
---
import { Playground, PropsTable } from 'docz'

import Search from './Search'

import '@cfg/scripts'
import '@cfg/style'

# Search
<PropsTable of={ Search }/>

## Basic usage

```
<Search/>
```

<Playground>
    <Search/>
</Playground>
