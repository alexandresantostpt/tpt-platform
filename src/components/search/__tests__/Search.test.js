import React from 'react'
import renderer from 'react-test-renderer'

import Search from '@components/search/Search'

describe('Tests for Search component', () => {
    it('Should rendering without crash ', () => {
        const tree = renderer.create(<Search />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
