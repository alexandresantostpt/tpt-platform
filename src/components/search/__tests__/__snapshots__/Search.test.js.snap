// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Search component Should rendering without crash  1`] = `
<div
  className="Search__StyledSearch-sc-1rsqafp-0 iZTykq"
>
  <input
    className="Search__StyledSearchInput-sc-1rsqafp-1 ddNcgu"
    onChange={[Function]}
    placeholder="Pesquise sua viagem"
  />
  <span
    className="icon icon-search Icon__StyledIcon-crr1cg-0 gEYSDH"
    disabled={false}
    fontSize="0.9rem"
    onClick={null}
  />
</div>
`;
